e2e/cypress/integration/team_settings/closed_team_invite_with_specific_domain_spec.js

```diff
@@ -0,0 +1,93 @@
+
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @team_settings
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+import {generateRandomUser} from '../../support/api/user';
+
+import {inviteUserByEmail, verifyEmailInviteAndVisitLink, typeUsernameAndPassword, verifyInitialTutorial} from './helpers';
+
+describe('Team Settings', () => {
+    let testTeam;
+    const user = generateRandomUser();
+    const {username, email, password} = user;
+    const emailDomain = 'sample.mattermost.com';
+
+    let isLicensed;
+
+    before(() => {
+        // # If the instance the test is running on is licensed, assign true to isLicensed variable
+        cy.apiGetClientLicense().then(({license}) => {
+            isLicensed = license.IsLicensed === 'true';
+        });
+
+        // # Disable LDAP and do email test if setup properly
+        cy.apiUpdateConfig({LdapSettings: {Enable: false}});
+        cy.apiEmailTest();
+
+        cy.apiInitSetup().then(({team}) => {
+            testTeam = team;
+            cy.visit(`/${team.name}/channels/town-square`);
+        });
+    });
+
+    it('MM-T386 Invite new user to closed team with \'Allow only users with a specific email domain to join this team\' set to \'sample.mattermost.com\'', () => {
+        // # Open 'Team Settings' modal
+        cy.findByLabelText('main menu').should('be.visible').click();
+        cy.findByText('Team Settings').should('be.visible').click();
+
+        // * Check that the 'Team Settings' modal was opened
+        cy.get('#teamSettingsModal').should('exist').within(() => {
+            // * Ensure that 'Allow any user with an account on this server' is set to 'No'
+            cy.get('#open_inviteDesc').should('have.text', 'No');
+
+            // # Click on the 'Allow only users with a specific email domain to join this team' edit button
+            cy.get('#allowed_domainsEdit').should('be.visible').click();
+
+            // # Set 'sample.mattermost.com' as the only allowed email domain and <save></save>
+            cy.wait(TIMEOUTS.HALF_SEC);
```

Comment:
I've tested several times without wait and I got a consistent passing result. Maybe we can safely remove this?
```suggestion
```